lib/web_ui/dev/steps/compile_tests_step.dart

```diff
@@ -122,30 +123,65 @@ Future<void> copySkiaTestImages() async {
 }
 
 Future<void> copyCanvasKitFiles() async {
+  // If CanvasKit has been built locally, use that instead of the CIPD version.
+  final io.File localCanvasKitWasm =
+      io.File(pathlib.join(environment.canvasKitOutDir.path, 'canvaskit.wasm'));
+  final bool builtLocalCanvasKit = localCanvasKitWasm.existsSync();
+
   final io.Directory canvasKitDir = io.Directory(pathlib.join(
     environment.engineSrcDir.path,
     'third_party',
     'web_dependencies',
     'canvaskit',
   ));
 
-  final Iterable<io.File> canvasKitFiles = canvasKitDir
-    .listSync(recursive: true, followLinks: true)
-    .whereType<io.File>();
+  Iterable<io.File> canvasKitFiles;
+  if (builtLocalCanvasKit) {
```

Comment:
This if/else can be combined with the if/else below. Unless I'm missing something, there no shared logic applied to `canvasKitFiles`. I think it can also reduce the amount of file path massaging done for the profile build.